Code/Framework/AzQtComponents/AzQtComponents/Components/Widgets/BreadCrumbs.cpp

```diff
@@ -134,6 +152,37 @@ namespace AzQtComponents
         return line;
     }
 
+    QSize BreadCrumbs::sizeHint() const
+    {
+        const QFontMetrics fm(m_label->font());
+        const qreal separatorWidth = fm.horizontalAdvance(g_plainTextSeparator);
+        // Icon adds the icon itself plus two spaces, see generateIconHtml()
+        const qreal iconWidth = g_iconWidth + fm.horizontalAdvance("\u00a0\u00a0");
+
+        // TODO(Qt 5.15.2): replace with:
+        // const QList<QStringView> fullPath = QStringView(m_currentPath).split(g_separator, Qt::SkipEmptyParts);
+        // to use views for avoiding allocations
+        const QStringList fullPath = m_currentPath.split(g_separator, Qt::SkipEmptyParts);
+
+        const int noSeparatorsWidth = AZStd::accumulate(fullPath.cbegin(), fullPath.cend(), 0,
+            [&fm](int val, const QString& pathSegment ){
+                return val + fm.horizontalAdvance(pathSegment);
+            }
+        );
+
+        // separators are there only if path has more than one segment
+        const qreal separatorsOnlyWidth = fullPath.size() > 1 ? (fullPath.size() - 1) * separatorWidth : .0;
+
+        const auto numIcons = AZStd::count_if(m_currentPathIcons.cbegin(), m_currentPathIcons.cend(), [this](const QString& iconPath){
+            // assume that icon will be there if there's a path for a given icon or we have a default one
+            return !iconPath.isEmpty() || !m_defaultIcon.isEmpty();
+        });
+
```

Comment:
Because `m_defaultIcon` doesn't change during the iteration, you can factor it out of the `count_if` call:
```suggestion
        const auto numIcons = !m_defaultIcon.isEmpty() ? m_currentPathIcons.size() : 
            AZStd::count_if(m_currentPathIcons.cbegin(), m_currentPathIcons.cend(), [](const QString& iconPath){
                // assume that icon will be there if there's a path for a given icon or we have a default one
                return !iconPath.isEmpty();
        });
```

Reply:
done